### DISCUSSION TITLE
Break Office keys

### DISCUSSION THREAD

**Discussion by Diablo-D3** [valinet/ExplorerPatcher#661] [Ideas]
Someone wrote this to kill Office keys dead, and seems to be the only working option I've seen:

https://github.com/anthonyheddings/OfficeKeyFix/blob/master/OfficeKeyFix.cpp

It binds all the known office keys, launches explorer so explorer can't bind them, then quits.

Seems like something ExplorerPatcher could easily integrate to remove yet another mal-feature Microsoft cursed us with.

**Comment by valinet** [comment]
What are “Office” keys?

**Comment by Diablo-D3** [comment]
Microsoft, on some of their keyboards, replaced the right Win key with an Office key, and since Windows cannot define more modifiers, their keyboards output Ctrl+Alt+Shift+Win (or what people commonly use for emulation of the Hyper key on Windows and other OSes that are modifier-deficient). They also added a dedicated emoji key (removing the menu key) because, apparently, pressing Win-Period is hard.

https://www.howtogeek.com/wp-content/uploads/2019/10/img_5db0ac14b839b.png

Windows does not actually check if you have a keyboard with an Office key, and because Explorer hijacks it during login, you can't otherwise bind it in any program. Because this idea is really dumb, the Surfaces do not have an Office key, only some variants of the Microsoft Ergonomic and some corporate Dells, HPs, etc. 

This mal-feature was added in the Win10 May 2019 update.

Example: Go press Ctrl-Alt-Shift-Win-L (which is the same as Office-L). T, Y, X, O, and W are also other keys. I won't spoil what they do, but it is depressing.

**Reply by valinet** [reply]
Implemented in latest pre-release (41.4).

**Reply by Diablo-D3** [reply]
Thanks man.

**Comment by riku22** [comment]
Hi.

Thanks for adding this feature.
I have one request, can't you also block when you press only `Ctrl`+`Alt`+`Shift`+`Windows`?
Currently, it seems to only block when the above keys are pressed together with another key, such as `W`.

**Reply by valinet** [reply]
`Ctrl`+`Alt`+`Shift`+`Windows` is not a valid combination, it's just a modifier. if you trap that combination, how would you differentiate between it and `Ctrl`+`Alt`+`Shift`+`Windows` + "some other key". I disabled the shortcuts assigned by `explorer`, you are now free to map `Ctrl`+`Alt`+`Shift`+`Windows` + "some other key" to whatever action you want using whatever software you want, as far as I checked.

**Comment by riku22** [comment]
Hello.

I see.
In my environment, pressing `Ctrl`+`Alt`+`Shift`+`Windows` by itself seems to open `https://www.office.com`.

**Reply by valinet** [reply]
Yeah, okay, disregard what I said, apparently you can `RegisterHotkey` using the modifiers alone, and it works. That's what `explorer` does. I excluded that as well, now you're free to assign `Ctrl`+`Alt`+`Shift`+`Windows` as well. A new pre-release will be compiled and published automatically in a few minutes.

It would have been nice if the original patch included this as well, just for the sake of completeness, but maybe the author did not know about this possibility either. Anyway, now it's fixed.

**Reply by valinet** [reply]
Yeah, it did the same for me. But if you look on the disassembly of `explorer`, as far as i remember, there are a few checks, depending on what is installed on the system, it may also launch the Office Hub or whatever. Office-releated things...

**Comment by riku22** [comment]
Thank you very much.

**Comment by Diablo-D3** [comment]
Wow, I didn't know about this one either. I wonder if this technically violates some internal guidelines at Microsoft, as pressing and releasing winkey should *always* fire the start menu no matter what. Office key is truly cursed.

**Reply by valinet** [reply]
If the API allows it, then it probably is okay. `RegisterHotkey` is a public API, every app could theoretically register that combination if `explorer` wouldn't be the first to keep it for itself. I mean, I think so, I haven't tried, I'd be curious to see whether with this patch one would be able to register that key combination in their own app....
